### PR TITLE
GSOC final pull request

### PR DESCRIPTION
**Pull request by datalogism**
Pushing all the work made during the GSOC

### CODE REVIEW

File: core/src/main/scala/org/dbpedia/extraction/util/MediawikiConnectorConfigured.scala [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    if(xmlAnswer.contains("error code=")) {
+      return Failure(new WebServiceException(pattern.findFirstMatchIn(xmlAnswer) match {
+        case Some(m) => m.group(2)
+        case None => "An unknown exception occurred while retrieving the source XML from the mediawiki API."
+      }))
+    }
+
+
+    // REDIRECT CASE
+    // Implemented but usefull ?
```
Comment (TallTed) [inline core/src/main/scala/org/dbpedia/extraction/util/MediawikiConnectorConfigured.scala:201]:
```suggestion
    // Implemented but useful?
```

File: dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* first clean your target directory with ``` mvn clean ``` in the root directory of DIEF
+* go to bash scripts via ``` cd /dump/src/test/bash  ```
+* OPTIONAL :  creating a new Wikipedia minidump sample with ``` bash create_custom_sample.sh -n $numberOfPage -l $lang -d $optionalDate ```
+* process sample of Wikipedia pages ``` bash Minidump_custom_sample.sh -f $filename/lst ```
+* Update your extraction language parameter regarding of your minidump sample in [extraction.nif.abstracts.properties](https://github.com/datalogism/extraction-framework/blob/gsoc-celian/dump/src/test/resources/extraction-configs/extraction.nif.abstracts.properties) and in [extraction.plain.abstracts.properties](https://github.com/datalogism/extraction-framework/blob/gsoc-celian/dump/src/test/resources/extraction-configs/extraction.plain.abstracts.properties)
+* Change the name of your log in the [**ExtractionTestAbstract.scala**](https://github.com/datalogism/extraction-framework/blob/gsoc-celian/dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.scala) file
+* rebuild the app with ``` mvn install ``` or just test it with ```  mvn test -Dtest="ExtractionTestAbstract2" ```
```
Comment (TallTed) [inline dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.md:16]:
```suggestion
1. Clean your target directory with `mvn clean` in the root directory of DIEF
1. Go to bash scripts via `cd /dump/src/test/bash`
1. OPTIONAL: Create a new Wikipedia minidump sample with `bash create_custom_sample.sh -n $numberOfPage -l $lang -d $optionalDate`
1. Process sample of Wikipedia pages `bash Minidump_custom_sample.sh -f $filename/lst`
1. Update the extraction language parameter for your minidump sample in [`extraction.nif.abstracts.properties`](https://github.com/datalogism/extraction-framework/blob/gsoc-celian/dump/src/test/resources/extraction-configs/extraction.nif.abstracts.properties) and in [`extraction.plain.abstracts.properties`](https://github.com/datalogism/extraction-framework/blob/gsoc-celian/dump/src/test/resources/extraction-configs/extraction.plain.abstracts.properties)
1. Change the name of your log in the [`ExtractionTestAbstract.scala`](https://github.com/datalogism/extraction-framework/blob/gsoc-celian/dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.scala) file
1. Rebuild the app with `mvn install`, or just test it with `mvn test -Dtest="ExtractionTestAbstract2"`
```

File: dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.scala [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      var numberOfFailedPagesOutOfMemoryError = 0
+      var numberOfFailedPagesNullPointerException = 0
+      var mapLocal= status
+
+      mapLocal += "language" -> lang.wikiCode.toString();
+
+      try {
+        val listFailedPages_ = records.listFailedPages(lang)
+        for( failed <- listFailedPages_) {
+          if(failed.toString().contains("Server returned HTTP response code: 429")){
```
Comment (TallTed) [inline dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.scala:123]:
```suggestion
          if (failed.toString().contains("Server returned HTTP response code: 429")){
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Reviewed commit by commit....

File: dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.scala [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+          }
+        }
+      } catch {
+        case e: Exception =>  None
+      }
+
+      mapLocal += "numberOfFailedPages429" -> numberOfFailedPages429.toString
+      mapLocal += "numberOfFailedPages503" -> numberOfFailedPages503.toString
+      mapLocal += "numberOfFailedPagesIOException" -> numberOfFailedPagesIOException.toString
+      mapLocal += "numberOfFailedPagesOutOfMemoryError" ->numberOfFailedPagesOutOfMemoryError.toString
```
Comment (TallTed) [inline dump/src/test/scala/org/dbpedia/extraction/dump/ExtractionTestAbstract.scala:146]: for code style consistency with neighboring lines
```suggestion
      mapLocal += "numberOfFailedPagesOutOfMemoryError" -> numberOfFailedPagesOutOfMemoryError.toString
```

File: core/src/main/scala/org/dbpedia/extraction/config/Config.scala [hunk 5]
```diff
@@ -259,18 +259,32 @@ class Config(val configPath: String) extends
...
+      useragent = this.getProperty("mwc-useragent", "anonymous").trim,
+      gzip = this.getProperty("mwc-gzip","false").trim.toBoolean,
+      retryafter = this.getProperty("mwc-retryafter", "false").trim.toBoolean,
+      accept = this.getProperty("mwc-accept", "text/html").trim,
+      charset = this.getProperty("mwc-charset", "utf-8").trim,
+      profile = this.getProperty("mwc-profile", "https://www.mediawiki.org/wiki/Specs/HTML/2.1.0").trim
     )
   } match{
     case Success(s) => s
     case Failure(f) => throw new IllegalArgumentException("Not all necessary parameters for the 'MediaWikiConnection' class were provided or could not be parsed to the expected type.", f)
```
Comment (TallTed) [inline core/src/main/scala/org/dbpedia/extraction/config/Config.scala:284]:
```suggestion
    case Failure(f) => throw new IllegalArgumentException("Some parameters necessary for the 'MediaWikiConnection' class were not provided or could not be parsed to the expected type.", f)
```

File: core/src/main/scala/org/dbpedia/extraction/config/Config.scala [hunk 6]
```diff
@@ -97,7 +97,7 @@ class Config(val configPath: String) extends
     * Number of parallel processes allowed. Depends on the number of cores, type of disk and IO speed
```
Comment (TallTed) [inline core/src/main/scala/org/dbpedia/extraction/config/Config.scala:97]:
```suggestion
    * Number of parallel processes allowed. Depends on the number of cores, type of disk, and IO speed
```

**Comment by datalogism** [comment]
> @celian_ringwald @datalogism — Are these all intermingled and/or interdependent? If not, it seems to me that each GSOC work item should have its own PR, rather than all being mushed into one. This will make all of them easier to review, among other things. (I recognize that it would be more work on your part, but I do think it would be worthwhile.)

Sorry @TallTed  I had no instructions about this, this is my first main pull request :s

**Review by TallTed — APPROVED** [review]

**Review by jlareck — CHANGES_REQUESTED** [review]

**Pull request merged**